### PR TITLE
SPM Prep – Move sources and tests in `Sources/` and `Tests/` respectively

### PR DESCRIPTION
**Pull request by mokagio**
What it says in the title. This PR only moved files but required a couple of adjustments in the Xcode project :

- Remove some "Group without folder" which would have made the setup harder to migrate
- Move the `WPKitLogging-` files out of the `Private` folder. The `.h` in particular is public in the project, but having it in that folder resulted in a validation failure at the CocoaPods level.

### Testing Details

See CI. 

---

- [ ] Please check here if your pull request includes additional test coverage. — N.A.
- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary. — N.A.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Podfile.lock [hunk 1]
```diff
@@ -56,4 +56,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 64af6d71574c7a92d01a9446aa874e066917ebe5
 
-COCOAPODS: 1.14.3
+COCOAPODS: 1.15.2
```
Comment (mokagio) [inline Podfile.lock:59]: I updated it as a preliminary, cover-all-bases, step before looking into a spec validation issue I incurred.

File: WordPressKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1298,13 +1298,11 @@
 		AB49D0B225D1B4D80084905B /* post-likes-failure.json */ = {isa = PBXFileReference; lastKnownFileType = text.json; path = "post-likes-failure.json"; sourceTree = "<group>"; };
 		ABD95B7E25DD6C4B00735BEE /* CommentServiceRemoteRESTLikesTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = CommentServiceRemoteRESTLikesTests.swift; sourceTree = "<group>"; };
 		ABD95B8425DD6DA200735BEE /* comment-likes-success.json */ = {isa = PBXFileReference; lastKnownFileType = text.json; path = "comment-likes-success.json"; sourceTree = "<group>"; };
-		B56D96B0202C9B7500485233 /* WordPressUI.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; path = WordPressUI.framework; sourceTree = BUILT_PRODUCTS_DIR; };
```
Comment (mokagio) [inline WordPressKit.xcodeproj/project.pbxproj:1301]: 👀 👀 👀 

What was this doing in here? A leftover from a long time ago when WordPressUI was part of the project?

File: WordPressKit.podspec [hunk 3]
```diff
@@ -18,9 +18,10 @@ Pod::Spec.new do |s|
   s.swift_version = '5.0'
 
   s.source        = { git: 'https://github.com/wordpress-mobile/WordPressKit-iOS.git', tag: s.version.to_s }
-  s.source_files  = 'WordPressKit/**/*.{h,m,swift}'
-  s.private_header_files = 'WordPressKit/Private/*.h'
-  s.header_dir = 'WordPressKit'
+  s.source_files  = 'Sources/**/*.{h,m,swift}'
```
Comment (mokagio) [inline WordPressKit.podspec:21]: I could have made this `Sources/WordPressKit/*.{h,m,swift}` but I thought I'd leave it looser for the moment. I'm not sure how/if to sort how subspecs as well as subpackages.

**Review by crazytonyli — APPROVED** [review]

**Pull request merged**